pkg/agent/util/winnet/net_windows.go

```diff
@@ -0,0 +1,886 @@
+	}
+	return parsed
+}
+
+func (h *Handle) NewNetNat(netNatName string, subnetCIDR *net.IPNet) error {
+	cmd := fmt.Sprintf(`Get-NetNat -Name %s | Select-Object InternalIPInterfaceAddressPrefix | Format-Table -HideTableHeaders`, netNatName)
+	if internalNet, err := runCommand(cmd); err != nil {
+		if !strings.Contains(err.Error(), "No MSFT_NetNat objects found") {
+			klog.ErrorS(err, "Failed to check the existing netnat", "name", netNatName)
+			return err
+		}
+	} else {
+		if strings.Contains(internalNet, subnetCIDR.String()) {
+			klog.V(4).InfoS("The existing netnat matched the subnet CIDR", "name", internalNet, "subnetCIDR", subnetCIDR.String())
+			return nil
+		}
+		klog.InfoS("Removing the existing NetNat", "name", netNatName, "internalIPInterfaceAddressPrefix", internalNet)
+		cmd = fmt.Sprintf("Remove-NetNat -Name %s -Confirm:$false", netNatName)
+		if _, err := runCommand(cmd); err != nil {
+			klog.ErrorS(err, "Failed to remove the existing netnat", "name", netNatName, "internalIPInterfaceAddressPrefix", internalNet)
+			return err
+		}
+	}
+	cmd = fmt.Sprintf(`New-NetNat -Name %s -InternalIPInterfaceAddressPrefix %s`, netNatName, subnetCIDR.String())
+	_, err := runCommand(cmd)
+	if err != nil {
+		klog.ErrorS(err, "Failed to add netnat", "name", netNatName, "internalIPInterfaceAddressPrefix", subnetCIDR.String())
+		return err
+	}
+	return nil
+}
+
+func (h *Handle) ReplaceNetNatStaticMapping(mapping *NetNatStaticMapping) error {
+	staticMappingStr, err := getNetNatStaticMapping(mapping)
+	if err != nil {
+		return err
+	}
+	parsed := parseGetNetCmdResult(staticMappingStr, 6)
+	if len(parsed) > 0 {
+		items := parsed[0]
+		if items[4] == mapping.InternalIP.String() && items[5] == strconv.Itoa(int(mapping.InternalPort)) {
+			return nil
+		}
+		firstCol := strings.Split(items[0], ";")
+		id, err := strconv.Atoi(firstCol[1])
+		if err != nil {
+			return err
+		}
+		if err := removeNetNatStaticMappingByID(mapping.Name, id); err != nil {
+			return err
+		}
+	}
+	return h.AddNetNatStaticMapping(mapping)
+}
+
+// getNetNatStaticMapping checks if a NetNatStaticMapping exists.
+func getNetNatStaticMapping(mapping *NetNatStaticMapping) (string, error) {
+	cmd := fmt.Sprintf("Get-NetNatStaticMapping -NatName %s", mapping.Name) +
+		fmt.Sprintf("|? ExternalIPAddress -EQ %s", mapping.ExternalIP) +
+		fmt.Sprintf("|? ExternalPort -EQ %d", mapping.ExternalPort) +
+		fmt.Sprintf("|? Protocol -EQ %s", mapping.Protocol) +
+		"| Format-Table -HideTableHeaders"
+	staticMappingStr, err := runCommand(cmd)
+	if err != nil && !strings.Contains(err.Error(), "No MSFT_NetNatStaticMapping objects found") {
+		return "", err
+	}
+	return staticMappingStr, nil
+}
+
+// AddNetNatStaticMapping adds a static mapping to a NAT instance.
+func (h *Handle) AddNetNatStaticMapping(mapping *NetNatStaticMapping) error {
+	cmd := fmt.Sprintf("Add-NetNatStaticMapping -NatName %s -ExternalIPAddress %s -ExternalPort %d -InternalIPAddress %s -InternalPort %d -Protocol %s",
```

Comment:
I guess this was just copied from the existing implementation, but do you know why sometimes we use regular strings (with double quotes) and sometimes we use raw strings (with backticks)?

Reply:
I am not sure about that. @wenyingd Do you know that reason for that? Thanks!

Reply:
I guess that double quotes are used to surround the command string in the original implementation only when there is no double quote in the string.